Dan6erbond/aPRAW
# Rework Listings

## Feature Summary
> Explain what's new in this pull request.

**Listings:**

 - Rework `class ListingGenerator` as `AsyncIterator` to perform requests.
 - Create `class streamable` decorator class to extend functions with `func.stream(wait_time)`.
 - Replace `class ListingGenerator` properties with functions returning new `class ListingGenerator`.

**Other:**

 - Update auto-property `aPRAWBase.fullname` to return ID if `kind` unknown.
 - Optimize imports and fix imports.
 - Fix JSON formatting in request dumps.

## Tests
> Describe tests if any were written.

 - Update `test_subreddit_wikipage_revert` and `test_subreddit_wikipage_hide` to work more efficiently with `class ListingGenerator` using list comprehension.
 - Fix `== True` check to `is True`.
 - Update `class TestListingGenerator` to be compatible with new `@streamable` decorator and use of functions instead of `class ListingGenerator`.
